src/core/lib/gprpp/notification.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_CORE_LIB_GPRPP_NOTIFICATION_H
+#define GRPC_CORE_LIB_GPRPP_NOTIFICATION_H
+
+#include <grpc/support/port_platform.h>
+
+#include "absl/time/clock.h"
+#include "absl/time/time.h"
+
+#include "src/core/lib/gprpp/sync.h"
+
+namespace grpc_core {
+
+// Polyfill for absl::Notification until we can use that type.
+class Notification {
```

Comment:
I thought about that, but then the only place we would be using this is on Mac where we have no visibility into stability right now, so if this broke due to a change we'd be completely blind.

Reply:
How about we add some tests that assert build failures on specific platforms? Then we'll know when we can drop the shims entirely, when those tests start failing.

Reply:
I don't know how that would work.

Current absl::Notification works everywhere (we're just adding this because we have flaky enough infrastructure that we can't convincingly prove that).

It would need to be a very high bar to add additional infrastructure to this shim.

Instead, once we've convinced ourselves that absl::Mutex is good everywhere, we should just delete this.